wls-gui-wahllokalsystem/src/components/wahlvorbereitung/TheUnguetilgeWahlscheineVerifyCard.vue

```diff
@@ -0,0 +1,192 @@
+        >{{ searchButtonLabel }}</v-btn
+      >
+      <base-button-refresh
+        :loading="ungueltigeWahlscheineIsLoading"
+        @click="onRefreshClicked"
+      />
+    </v-card-actions>
+  </v-card>
+</template>
+
+<script setup lang="ts">
+import type { UngueltigerWahlschein } from "@/types/wahlbezirk/UngueltigerWahlschein.ts";
+import type { ShallowRef } from "vue";
+
+import { storeToRefs } from "pinia";
+import { computed, ref, useTemplateRef } from "vue";
+import {
+  VBtn,
+  VCard,
+  VCardActions,
+  VCardText,
+  VCardTitle,
+  VForm,
+  VImg,
+  VNumberInput,
+} from "vuetify/components";
+
+import wahlscheinExampleImage from "@/assets/previewWahlscheinnummerOnWahlschein.png";
+import BaseButtonRefresh from "@/components/common/buttons/BaseButtonRefresh.vue";
+import BaseInputFeedbackCard from "@/components/common/cards/BaseInputFeedbackCard.vue";
+import { useWahlbezirkStore } from "@/stores/wahlbezirkStore.ts";
+import { InputFeedbackTypeEnum } from "@/types/common/InputFeedbackTypeEnum.ts";
+import { MAX_NUMBER, MIN_NUMBER, REQUIRED } from "@/util/rules.ts";
+
+const isFormValid = ref<boolean | null>(null);
+const isSearchButtonDisabled = computed(() => isFormValid.value !== true);
+const wahlscheinValidationForm = useTemplateRef(
+  "wahlscheinValidationForm"
+) as Readonly<ShallowRef<InstanceType<typeof VForm>>>;
+
+const {
+  getUngueltigerWahlscheinByWahlscheinnummer,
+  loadUngueltigeWahlscheine,
+} = useWahlbezirkStore();
+const {
+  ungueltigeWahlscheineIsLoading,
+  ungueltigeWahlscheineIsEmpty,
+  ungueltigeWahlscheineLoadingFailed,
+} = storeToRefs(useWahlbezirkStore());
+
+const wahlscheinnummer = ref<null | number>(null);
+const ungueltigerWahlschein = ref<null | undefined | UngueltigerWahlschein>(
+  undefined
+);
+
+const feedbackNoDataAvailableIsVisible = computed(
+  () =>
+    ungueltigeWahlscheineIsEmpty.value &&
+    !ungueltigeWahlscheineLoadingFailed.value &&
+    !ungueltigeWahlscheineIsLoading.value
+);
+const feedbackLoadingFailedIsVisible = computed(
+  () =>
+    ungueltigeWahlscheineLoadingFailed.value &&
+    !ungueltigeWahlscheineIsLoading.value
+);
+const feedbackWahlscheinIsGueltigIsVisible = computed(
+  () => ungueltigerWahlschein.value === null
+);
+const feedbackWahlscheinIsUngueltigIsVisible = computed(
+  () => ungueltigerWahlschein.value
+);
+const searchButtonLabel = computed(() =>
+  ungueltigerWahlschein.value === undefined ? "Suchen" : "Suche zurücksetzen"
+);
+const titleFeedbackWahlscheinUngueltig = computed(() => {
+  return `Wahlschein ${ungueltigerWahlschein.value?.wahlscheinnummer ?? ""} für ${ungueltigerWahlschein.value?.vorname} ${ungueltigerWahlschein.value?.familienname} ist ungültig`;
+});
+
+function onRefreshClicked() {
+  loadUngueltigeWahlscheine();
+}
+
+function onSearchClicked() {
+  if (ungueltigerWahlschein.value !== undefined) {
+    resetUngueltigerWahlschein();
+    wahlscheinValidationForm.value.reset();
+  } else if (wahlscheinnummer.value !== null) {
```

Comment:
wahlschein nummer ist immer Größer null, daher sollte das auch gehen
```suggestion
  } else if (wahlscheinnummer.value) {
```

Reply:
Theoretisch ja, aber ich denke es ist besser mit `null` zu arbeiten. Es könnte sein das zu Testzwecke `0` vergeben wird. Und fachliche sind alle Nummer möglich.